### PR TITLE
Pseudo ppg label function

### PR DESCRIPTION
**Pull request by girishvn**
**Train: PURE, Test: UBFC-rPPG, Model: TSCAN**
FFT MAE (FFT Label): 1.3392857142857142 +/- 0.42190125596543704
FFT RMSE (FFT Label): 3.044620560179667 +/- 3.5261431285376057
FFT MAPE (FFT Label): 1.5575960591866085 +/- 0.5049767631839935
FFT Pearson (FFT Label): 0.988079459671649 +/- 0.02434079978913088

**Train: PURE Pseudo Labels, Test: UBFC-rPPG GT Labels, Model: TSCAN**
FFT MAE (FFT Label): 9.375 +/- 2.5283082691165326
FFT RMSE (FFT Label): 18.87773870342228 +/- 142.48195440534639
FFT MAPE (FFT Label): 13.307904748480684 +/- 4.077642607327307
FFT Pearson (FFT Label): 0.582888550823001 +/- 0.12847576982834757
FFT SNR (FFT Label): -1.8506740608244514 +/- 1.7017036801128422

### CODE REVIEW

**Comment by yahskapar** [comment]
Does the current logic and the absence of a USE_PSUEDO_PPG_LABEL for the UNSUPERVISED config portion of `config.py` potentially throw an error when any of the dataloaders are used for unsupervised methods? LGTM aside from that.

File: dataset/data_loader/UBFCrPPGLoader.py [hunk 1]
```diff
@@ -79,8 +80,13 @@ def preprocess_dataset_subprocess(self, data_dirs, config_preprocess, i, file_li
...
             raise ValueError(f'Unsupported DATA_AUG specified for {self.dataset_name} dataset! Received {config_preprocess.DATA_AUG}.')
-        bvps = self.read_wave(
-            os.path.join(data_dirs[i]['path'],"ground_truth.txt"))
+
+        # Read Labels
+        if config_preprocess.USE_PSUEDO_PPG_LABEL:
+            bvps = self.generate_pos_psuedo_labels(frames, fs=self.config_data.FS)
+        else:
+            bvps = self.read_wave(
+                os.path.join(data_dirs[i]['path'],"ground_truth.txt"))
```
Comment (yahskapar) [inline dataset/data_loader/UBFCrPPGLoader.py:89]: Maybe for the above code here and in other loaders, you could try the following after exposing the TOOLBOX_MODE through the class init for each loader:

```
# Read Labels
if config.TOOLBOX_MODE == "unsupervised_method":
        bvps = self.read_wave(
                        os.path.join(data_dirs[i]['path'],"ground_truth.txt"))
elif config_preprocess.USE_PSUEDO_PPG_LABEL:
        bvps = self.generate_pos_psuedo_labels(frames, fs=self.config_data.FS)
else:
        bvps = self.read_wave(
                os.path.join(data_dirs[i]['path'],"ground_truth.txt"))
```

**Review by yahskapar — CHANGES_REQUESTED** [review]

**Review by yahskapar — APPROVED** [review]
LGTM. Consider shortening the d9ff8b27b114386a3ed4ad893e47901f127079e6 commit message.

**Comment by girishvn** [comment]
Edited commit message. Merging.

**Pull request merged**